packages/blade/src/components/ProgressBar/ProgressBar.tsx

```diff
@@ -0,0 +1,185 @@
+   */
+  isIndeterminate?: boolean;
+  /**
+   * Sets whether or not to show the progress percentage for the progress bar. Percentage is hidden by default for the `meter` variant.
+   * @default false
+   */
+  hidePercentage?: boolean;
+};
+
+type ProgressBarMeterProps = ProgressBarCommonProps & {
+  /**
+   * Sets the variant to be rendered for thr progress bar.
+   * @default 'progress'
+   */
+  variant?: Extract<ProgressBarVariant, 'meter'>;
+  /**
+   * Sets whether the progress bar is in an indeterminate state.
+   * @default false
+   */
+  isIndeterminate?: undefined;
+  /**
+   * Sets whether or not to show the progress percentage for the progress bar. Percentage is hidden by default for the `meter` variant.
+   * @default false
+   */
+  hidePercentage?: undefined;
+};
+
+type ProgressBarProps = ProgressBarProgressProps | ProgressBarMeterProps;
+
+type ProgressBarUnfilledProps = {
+  backgroundColor: string;
+};
+
+const ProgressBarUnfilled = styled(Box)<ProgressBarUnfilledProps>(({ backgroundColor }) => ({
+  backgroundColor,
+}));
+
+const getProgress = (value: number): number => {
+  return Math.floor(Math.min(100, Math.max(0, value)));
+};
+
+const progressBarHeight: Record<NonNullable<ProgressBarCommonProps['size']>, 2 | 4> = {
+  small: 2,
+  medium: 4,
+};
+
+const ProgressBar = ({
+  accessibilityLabel,
+  contrast = 'low',
+  intent,
+  isIndeterminate,
+  label,
+  hidePercentage = false,
+  size = 'small',
+  value = 0,
+  variant = 'progress',
+}: ProgressBarProps): ReactElement => {
+  console.log(isIndeterminate); // will be in follow up PR
+
+  if (variant === 'meter' && isIndeterminate) {
+    console.warn(`[Blade: ProgressBar]: Cannot set 'isIndeterminate' when 'variant' is 'meter'.`);
+  }
+
+  if (variant === 'meter' && hidePercentage) {
+    console.warn(
+      `[Blade: ProgressBar]: Cannot set 'hidePercentage' when 'variant' is 'meter'. Percentage is always hidden for meter.`,
+    );
+  }
+
+  const { theme } = useTheme();
+  const unfilledBackgroundColor = theme.colors.brand.gray.a100[`${contrast}Contrast`];
+  const filledBackgroundColor = intent
+    ? theme.colors.feedback.background[intent].highContrast
+    : theme.colors.brand.primary[500];
+  const hasLabel = label && label.trim()?.length > 0;
+  const isMeter = variant === 'meter';
+  const progressValue = getProgress(value);
+  const id = useId(variant);
+
+  return (
+    <Box>
+      <Box
+        display="flex"
+        flexDirection="row"
+        justifyContent={hasLabel ? 'space-between' : 'flex-end'}
+      >
+        {hasLabel ? (
+          <FormLabel as="label" htmlFor={id} contrast={contrast}>
+            {label}
+          </FormLabel>
+        ) : null}
+        {hidePercentage || isMeter ? null : (
+          <Box marginBottom="spacing.2">
+            <Text
+              type="subdued"
+              variant="body"
+              contrast={contrast}
+              size="small"
+            >{`${progressValue}%`}</Text>
```

Comment:
Just confirming, do we want to render the UI even if it's `0%`?

Reply:
Yup!